core/web/migrate_controller.go

```diff
@@ -0,0 +1,207 @@
+package web
+
+import (
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/smartcontractkit/chainlink/core/logger"
+
+	"github.com/smartcontractkit/chainlink/core/store/orm"
+
+	"github.com/gin-gonic/gin"
+	"github.com/pkg/errors"
+	"github.com/smartcontractkit/chainlink/core/adapters"
+	"github.com/smartcontractkit/chainlink/core/services/chainlink"
+	"github.com/smartcontractkit/chainlink/core/services/job"
+	"github.com/smartcontractkit/chainlink/core/services/pipeline"
+	"github.com/smartcontractkit/chainlink/core/store/models"
+	webpresenters "github.com/smartcontractkit/chainlink/core/web/presenters"
+	"gonum.org/v1/gonum/graph/encoding/dot"
+	"gopkg.in/guregu/null.v4"
+)
+
+var (
+	ErrInvalidInitiatorType = errors.New("invalid initiator type")
+)
+
+type MigrateController struct {
+	App chainlink.Application
+}
+
+// Creates a v2 job from a v1 job ID.
+// Example:
+//  "POST <application>/migrate/123e4567-e89b-12d3-a456-426614174000"
+// Where "123e4567-e89b-12d3-a456-426614174000" is a v1 job ID.
```

Comment:
Presumably this is irreversible? Is there an easy way for a node operator to "undo" a migration?

Reply:
No easy way - it's roll forward

Reply:
OK as long as we explain this in the Changelog